src/components/MultiColumnMeasures.tsx

```diff
@@ -1,81 +1,85 @@
 import Instruction from './Instruction';
+import MultiColumnMeasureTabs from './MultiColumnMeasureTabs';
 import MultiColumnMeasuresCard from './MultiColumnMeasuresCard';
+import MultiColumnMeasuresColumnsSidebar from './MultiColumnMeasuresColumnsSidebar';
 
 function MultiColumnMeasures() {
-  const theme = useTheme();
-  const [activeTab, setActiveTab] = useState(0);
-  const updateColumnStandardizedVariable = useDataStore(
-    (state) => state.updateColumnStandardizedVariable
+  const {
+    variables: multiColumnVariables,
+    activeTab,
+    activeVariable,
+    setActiveTab,
+  } = useMultiColumnMeasureTabs();
+  const activeVariableId = activeVariable?.id || '';
+  const persistedCards = usePersistedMultiColumnCards(activeVariableId);
+  const { draftMeasureCards, createDraft, updateDraftTerm, removeDraft, hasDraft } =
+    useMultiColumnMeasureDraftCard(activeVariableId);
+  const combinedCards = [...persistedCards, ...draftMeasureCards];
+  const { cardData, variableAllMappedColumns } = useMultiColumnMeasureCardData(
+    activeVariableId,
+    combinedCards,
+    persistedCards
   );
-
-  const stateManager = useMultiColumnMeasuresState();
-  const { multiColumnVariables, columns } = useMultiColumnMeasuresData();
-  const { activeVariableTab, currentVariableColumns, currentTermCards, variableAllMappedColumns } =
-    useActiveVariableData(multiColumnVariables, activeTab);
-
-  const handleTabChange = (_: React.SyntheticEvent, newValue: number) => {
+  const columns = useColumns();
+  const actions = useFreshDataActions();
+  const currentVariableColumns = activeVariable
+    ? Object.entries(columns)
+        .filter(([_, column]) => column.standardizedVariable === activeVariable.id)
+        .map(([id, column]) => ({ id, header: column.name }))
+    : [];
+  const canAddDraft = !hasDraft;
+
+  const handleTabChange = (_: SyntheticEvent, newValue: number) => {
     setActiveTab(newValue);
   };
 
   const handleAddNewCard = () => {
-    if (activeVariableTab) {
-      stateManager.addTermCard(activeVariableTab.identifier);
-    }
+    createDraft();
   };
 
-  const handleTermSelect = (cardId: string, term: MultiColumnMeasuresTerm | null) => {
-    if (activeVariableTab) {
-      stateManager.updateTermInCard(activeVariableTab.identifier, cardId, term);
-    }
+  const handleTermSelect = (cardId: string, termId: string | null) => {
+    if (!termId) return;
+    updateDraftTerm(cardId, termId);
   };
 
-  const handleColumnSelect = (cardId: string, columnId: string | null) => {
-    if (columnId && activeVariableTab) {
-      stateManager.addColumnToCard(activeVariableTab.identifier, cardId, columnId);
-    }
+  const handleColumnSelect = (cardId: string, termId: string, columnId: string | null) => {
+    if (!columnId) return;
+    actions.userUpdatesColumnIsPartOf(columnId, termId);
+    actions.userUpdatesMultiColumnMeasureCards(termId, true);
+    removeDraft(cardId);
```

Comment:
also this should have already happened when user selected Collection StandardizedTerm, i.e. would be responsibility of the handler currently called `handleTermSelect`